incubator/group/types.proto

```diff
@@ -304,3 +304,82 @@ message GenesisState {
     option (gogoproto.goproto_stringer) = false;
     Params Params = 1 [(gogoproto.nullable)=false];
 }
+
+//
+// Queries
+//
+service Query {
+    rpc GetGroupMetadata(GroupMetadataRequest) returns (GroupMetadata);
+    rpc GetGroupAccountMetadata(GroupAccountMetadataRequest) returns (AnyGroupAccountMetadata);
+    rpc GetGroupsByMember(ByAddressRequest) returns (GroupsList);
+    rpc GetGroupsByOwner(ByAddressRequest) returns (GroupsList);
+    rpc GetGroupAccountsByGroup(ByGroupRequest) returns (GroupAccountsList);
+    rpc GetGroupAccountsByOwner(ByAddressRequest) returns (GroupAccountsList);
+    rpc GetProposal(ProposalRequest) returns (AnyProposal);
+    rpc GetProposalsByGroupAccount(ByGroupAccountRequest) returns (ProposalList);
+    rpc GetVotes(VotesRequest) returns (VoteList);
+}
+
+message GroupMetadataRequest {
+    uint64 group = 1; // GroupID
+}
+
+message GroupAccountMetadataRequest {
+    bytes group_account = 1;
+}
+
+// AnyGroupAccountMetadata wraps its decision policy in Any for a consistent querier interface
+message AnyGroupAccountMetadata {
+    GroupAccountMetadataBase base = 1 [(gogoproto.nullable) = false];
+    google.protobuf.Any decision_policy = 2;
+}
+
+message ByAddressRequest {
+    bytes address = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+    uint64 offset = 2;
```

Comment:
I have seen this in sdk code as well. General speaking it is an expensive operation to iterate through the whole KV store to count for an offset.